pkg/controllers/state/cluster.go

```diff
@@ -497,7 +544,10 @@ func (c *Cluster) populateResourceRequests(ctx context.Context, n *StateNode) er
 		if err := n.updateForPod(ctx, c.kubeClient, pod); err != nil {
 			return err
 		}
-		c.cleanupOldBindings(pod)
+		// if the pod binding is new, cleanup old bindings
+		if oldNodeName, bindingKnown := c.bindings[client.ObjectKeyFromObject(pod)]; !bindingKnown || oldNodeName != pod.Spec.NodeName {
```

Comment:
Is this just a refactor or is there a behavior change here? Seems like we just pulled the `if` conditional out?